src/deepsparse/pipelines/embedding_extraction.py

```diff
@@ -0,0 +1,298 @@
+from deepsparse.log import get_main_logger
+from deepsparse.transformers.helpers import truncate_transformer_onnx_model
+
+
+__all__ = [
+    "EmbeddingExtractionOutput",
+    "EmbeddingExtractionPipeline",
+]
+
+_LOGGER = get_main_logger()
+
+
+class EmbeddingExtractionOutput(BaseModel):
+    """
+    Schema for embedding_extraction pipeline output. Values are in batch order
+    """
+
+    embeddings: Union[List[List[float]], List[numpy.ndarray]] = Field(
+        description="The output of the model which is an embedded "
+        "representation of the input"
+    )
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class ExtractionStrategy(str, Enum):
+    """
+    Schema for supported extraction strategies
+    """
+
+    per_token = "per_token"
+    reduce_mean = "reduce_mean"
+    reduce_max = "reduce_max"
+    cls_token = "cls_token"
+
+    @classmethod
+    def to_list(cls) -> List[str]:
+        return cls._value2member_map_
+
+
+@Pipeline.register(
+    task="embedding_extraction",
+    task_aliases=[],
+    default_model_path="",  # TODO determine what goes here
+)
+class EmbeddingExtractionPipeline(Pipeline):
+    """
+    LIFECYCLE:
+
+    Initialize:
+        1. create base pipeline stopping before engine initialization
+        2. extract onnx file path from base pipeline and truncate
+        3. finish base pipeline engine initialization
+
+    __call__:
+        1. input schema defers to base pipeline
+        2. pre-processing defers to base pipeline
+        3. engine_forward runs truncated model
+        4. post-processing runs embedding aggregation
+        5. output schema contains the possibly aggregated embeddings
+    """
+
+    def __init__(
+        self,
+        *,
+        base_task: str,
+        emb_extraction_layer: Union[int, str, None] = -1,
+        model_size: int = 768,
+        extraction_strategy: ExtractionStrategy = "per_token",
+        return_numpy: bool = True,
+        **base_pipeline_args,
+    ):
+        self._base_task = base_task
+        self._emb_extraction_layer = emb_extraction_layer
+        self._model_size = model_size
+        self._extraction_strategy = extraction_strategy
+        self._return_numpy = return_numpy
+
+        if self._extraction_strategy not in ExtractionStrategy.to_list():
+            raise ValueError(
+                f"Unsupported extraction_strategy {self._extraction_strategy}"
+            )
+
+        self.base_pipeline = Pipeline.create(
+            task=base_task,
+            _delay_engine_initialize=True,  # engine initialized after model truncate
+            **base_pipeline_args,
+        )
+
+        pipeline_keyword_names = {
+            "model_path",
+            "engine_type",
+            "batch_size",
+            "num_cores",
+            "scheduler",
+            "input_shapes",
+            "alias",
+            "context",
+            "executor",
+            "_delay_engine_initialize",
+        }
```

Comment:
let's use `inspect.signature(Pipeline).parameters.keys()` instead - if the kwargs were to change this will break silently